src/Simplex/Messaging/Agent.hs

```diff
@@ -1944,16 +2056,28 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
         -- mark queue as Secured and to start sending messages to it
         qUseMsg :: NonEmpty ((SMPServer, SMP.SenderId), Bool) -> Connection 'CDuplex -> m ()
         -- NOTE: does not yet support the change of the primary status during the rotation
-        qUseMsg ((addr, _primary) :| _) (DuplexConnection _ _ sqs) =
-          case findQ addr sqs of
-            Just sq' -> do
-              logServer "<--" c srv rId $ "MSG <QUSE> " <> logSecret (snd addr)
-              withStore' c $ \db -> setSndQueueStatus db sq' Secured
-              let sq'' = (sq' :: SndQueue) {status = Secured}
-              -- sending QTEST to the new queue only, the old one will be removed if sent successfully
-              void $ enqueueMessages c cData [sq''] SMP.noMsgFlags $ QTEST [addr]
-              notify . SWITCH QDSnd SPConfirmed $ connectionStats conn
-            _ -> qError "QUSE: queue address not found in connection"
+        qUseMsg ((addr, _primary) :| _) (DuplexConnection _ rqs sqs) =
+          case switchingSQ sqs of
```

Comment:
searching for the switching queue seems a wrong approach, it must be determined by the context.

There are two options:
1. only primary queue can be switched - in this case it's always the first queue, and we just need to check it's switching status, not search for this queue. It also creates a potential vulnerability, as the message can arrive to another queue - it should be just ignored in this case.
2. we want data model that supports redundancy and multiple queues can be switched concurrently.

In either case, this should not be determined by queue switching status, but should be contextual.